Task_1.py

```diff
@@ -1,160 +1,165 @@
             count_unanalyzed_2 += 1
-    comment_2 = [
-        str(count_unanalyzed_2) + ' - the number of records that cannot be processed '
-                                'to get the application which received the most useless message']
-    comment.append(comment_2)
-
-    count_unanalyzed_3 = len(main_dict) - len(set(analyzed_data))
-    comment_3 = [
-        str(count_unanalyzed_3) + ' or ' + str(round(count_unanalyzed_3/len(main_dict) * 100)) + '% ' +
-        ' - the number of records that cannot be processed to get the shortest interval between '
-        'ratings of one user (among all users) and the length of both messages which create this interval;']
-    comment.append(comment_3)
+    comment.append([count_unanalyzed_2, ' - to get the application which received the most useless message'])
+
+    count_unanalyzed_3 = len(main_dict) - analyzed_data
+    comment.append([str(count_unanalyzed_3) + ' or ' + str(round(count_unanalyzed_3/len(main_dict) * 100)) + '% ',
+                    ' - to get the shortest interval between ratings of one user (among all users) '
+                    'and the length of both messages which create this interval;'])
 
     count_unanalyzed_4 = count_unanalyzed_2
     for item in main_dict.values():
         if item['helpful'][1] == 0:
             count_unanalyzed_4 += 1
-    comment_4 = [
-        str(count_unanalyzed_4) + ' or ' + str(round(count_unanalyzed_4/len(main_dict) * 100)) + '% ' +
-        ' - the number of records that cannot be processed to get '
-        'the application which received the most useless message']
-    comment.append(comment_4)
+    comment.append([str(count_unanalyzed_4) + ' or ' + str(round(count_unanalyzed_4/len(main_dict) * 100)) + '% ',
+                    ' - to get the application which received the most useless message'])
     return comment
 
 
-# create file general-stats.cvs containing information above:
-def file_create(data):
-    path = os.path.dirname(os.path.abspath(__file__))
-    try:
-        os.mkdir(os.path.join(path, 'resulting data'))
-    except:
-        pass
-    with open(os.path.join(path, 'resulting data', 'general-stats.cvs'), "w", newline='') as csv_file:
-        writer = csv.writer(csv_file, delimiter='\t')
-        for line in data:
-            writer.writerow(line)
-
 if __name__ == "__main__":
     main_dict = get_data.open_gzip()
-    common_result = []
     analyzed_data = []
-    common_result.append(avg_rating(main_dict))
-    common_result.append(best_comment(main_dict))
-    common_result.append(nearest_review(main_dict))
-    common_result.append(bad_comment(main_dict))
-    common_result.append(nonanalys_data(main_dict, analyzed_data))
-    file_create(common_result)
-
+    file_name = 'general-stats.cvs'
+    file_create().save_file(file_name, avg_rating(main_dict))
+    file_create().save_file(file_name, best_comment(main_dict), 'a')
```

Comment:
Import save_file func directly, not need to create uselessly object each time.

Reply:
Will do!